kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/bootstrap/FilterChainFactoryTest.java

```diff
@@ -238,4 +247,57 @@ private ListAssert<FilterAndInvoker> assertFiltersCreated(List<FilterDefinition>
         List<FilterAndInvoker> filters = filterChainFactory.createFilters(context);
         return assertThat(filters).hasSameSizeAs(filterDefinitions);
     }
+
+    @Test
+    void shouldPropagateFilterInitializeException() {
+        // Given
+        final FilterDefinition requiredConfig = new FilterDefinition(FlakyFactory.class.getName(),
+                new FlakyConfig("foo", null, null));
+        List<FilterDefinition> list = List.of(requiredConfig);
+
+        // When
+
+        // Then
+        assertThatThrownBy(() -> new FilterChainFactory(pfr, list))
+                .isExactlyInstanceOf(PluginConfigurationException.class)
+                .cause()
+                .isExactlyInstanceOf(RuntimeException.class)
+                .hasMessage("foo");
+    }
+
+    @Test
+    void shouldPropagateFilterCreateException() {
+        // Given
+        final FilterDefinition requiredConfig = new FilterDefinition(FlakyFactory.class.getName(),
+                new FlakyConfig(null, "foo", null));
+        List<FilterDefinition> list = List.of(requiredConfig);
+        NettyFilterContext context = new NettyFilterContext(eventLoop, pfr);
+
+        try (var fcf = new FilterChainFactory(pfr, list)) {
+            // When
+
+            // Then
+            assertThatThrownBy(() -> fcf.createFilters(context))
+                    .isExactlyInstanceOf(PluginConfigurationException.class)
+                    .cause()
+                    .isExactlyInstanceOf(RuntimeException.class)
+                    .hasMessage("foo");
+        }
+    }
+
+    @Test
+    void shouldPropagateFilterCloseException() {
+        // Given
+        final FilterDefinition requiredConfig = new FilterDefinition(FlakyFactory.class.getName(),
+                new FlakyConfig(null, null, "foo"));
+        List<FilterDefinition> list = List.of(requiredConfig);
+        try (var fcf = new FilterChainFactory(pfr, list)) {
+            // When
+
+            // Then
+            assertThatThrownBy(fcf::close)
+                    .isExactlyInstanceOf(RuntimeException.class)
+                    .hasMessage("foo");
+        }
+    }
```

Comment:
I think these are a necessary but not sufficient. I think we need to validate that if one factory in the chain throws the other factories are closed as appropriate.

Reply:
Added some stronger assertions.